### PR TITLE
MODQM-27 - quickMARC record's edit status

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODQM-27](https://issues.folio.org/browse/MODQM-27) - quickMARC record's edit status

## Purpose
In order to apply source-record states storing mechanism one need to implement corresponding changes on quickMARC.

## Approach
* added status-related fields into QuickMarcJson schema (recordState, updatedDate)
* status-related fields are updated on each GET record request 
* updated test data

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [x] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by khandramai** [comment]
Looks like examples (with actual changes) don't correspond QuickMarcJson schemas. Could you please doublecheck examples against schema https://www.jsonschemavalidator.net.
Screen Shot 2020-07-17 at 10 43 07 AM

File: ramls/schemas/quick_marc_json.json [hunk 1]
```diff
@@ -66,7 +66,28 @@
...
+    },
+    "updateInfo": {
+      "description": "Information related to record update",
+      "type": "object",
+      "properties": {
+        "recordState": {
+          "description": "State of source record in QM workflow",
+          "type": "string"
+        },
+        "updatedDate": {
```
Comment (khandramai) [inline ramls/schemas/quick_marc_json.json:78]: Maybe it makes sense to name this updateDate?

File: ramls/examples/quick_marc_json.sample [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+  "instanceId": "b9a5f035-de63-4e2c-92c2-07240c89b817",
+  "suppressDiscovery": false,
   "leader": "01587ccm a2200361   4500",
-  "records": [
+  "fields": [
     {
       "tag": "001",
-      "content": "393893"
+      "content": "393893",
+      "indicators": []
```
Comment (khandramai) [inline ramls/examples/quick_marc_json.sample:11]: Do we need indicators for control fields?

File: src/test/resources/mockdata/quick-marc-json/books.json [hunk 3]
```diff
@@ -1,91 +1,66 @@
...
-}
+    "indicators" : [ ]
+  }, {
+    "tag" : "010",
+    "content" : "$a    55001156/M ",
+    "indicators" : [ " ", " " ]
+  } ],
+  "updateInfo" : {
+    "recordState" : "ACTUAL",
+    "updatedDate" : 1594901616879
```
[the remark this answers is not in the document]
Comment (siarhei-charniak) [reply src/test/resources/mockdata/quick-marc-json/books.json]: fixed

**Pull request merged**